tests/perf/workflows/workflow_test.go

```diff
@@ -106,122 +118,94 @@ func addTestResults(t *testing.T, testName string, testAppName string, result *l
-
-				// Stop the workflow runtime
-				url = fmt.Sprintf("http://%s/shutdown-workflow-runtime", externalURL)
-				_, err = utils.HTTPGet(url)
-				require.NoError(t, err, "error shutdown workflow runtime")
-
-			})
-		}
+func TestWorkFlowPerf(t *testing.T) {
+	tcs := []struct {
+		name         		string
+		rateCheck       	string
+		enableMemoryCheck	bool
+	}{
+		{
+			name:         "average_load",
+			rateCheck:    "rate==1",
+			enableMemoryCheck: true,
+
+		},
+		{
+			name:         "comprehensive_load",
+			rateCheck:    "rate==1",
+			enableMemoryCheck: false,
+		},
 	}
-	err := table.Flush()
-	require.NoError(t, err, "error storing test results")
-}
-
-// Runs tests for `sum_series_wf` with constant VUs
-func TestWorkflowWithConstantVUs(t *testing.T) {
-	workflowName := "sum_series_wf"
-	inputs := []string{"100"}
-	scenarios := []string{"t_30_300", "t_30_300", "t_30_300"}
-	rateChecks := [][]string{{"rate==1", "rate==1", "rate==1"}}
-	testWorkflow(t, workflowName, testAppNames[0], inputs, scenarios, rateChecks, false, false)
-}
-
-func TestWorkflowWithConstantIterations(t *testing.T) {
-	workflowName := "sum_series_wf"
-	inputs := []string{"100"}
-	scenarios := []string{"t_30_300", "t_60_300", "t_90_300"}
-	rateChecks := [][]string{{"rate==1", "rate==1", "rate==1"}}
-	testWorkflow(t, workflowName, testAppNames[0], inputs, scenarios, rateChecks, true, false)
-}
-
-// Runs tests for `sum_series_wf` with Max VUs
-func TestSeriesWorkflowWithMaxVUs(t *testing.T) {
-	workflowName := "sum_series_wf"
-	inputs := []string{"100"}
-	scenarios := []string{"t_350_1400"}
-	rateChecks := [][]string{{"rate==1"}}
-	testWorkflow(t, workflowName, testAppNames[0], inputs, scenarios, rateChecks, true, false)
-}
-
-// Runs tests for `sum_parallel_wf` with Max VUs
-func TestParallelWorkflowWithMaxVUs(t *testing.T) {
-	workflowName := "sum_parallel_wf"
-	inputs := []string{"100"}
-	scenarios := []string{"t_110_440"}
-	rateChecks := [][]string{{"rate==1"}}
-	testWorkflow(t, workflowName, testAppNames[0], inputs, scenarios, rateChecks, true, false)
-}
-
-// Runs tests for `state_wf` with different Payload
-func TestWorkflowWithDifferentPayloads(t *testing.T) {
-	workflowName := "state_wf"
-	scenarios := []string{"t_30_300"}
-	inputs := []string{"10000", "50000", "100000"}
-	rateChecks := [][]string{{"rate==1"}, {"rate==1"}, {"rate==1"}}
-	testWorkflow(t, workflowName, testAppNames[0], inputs, scenarios, rateChecks, true, true)
-}
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			table := summary.ForTest(t)
+			subTestName := "[" + tc.name +"]: "
+			// Get the ingress external url of test app
+			log.Println("acquiring app external URL")
+			externalURL := tr.Platform.AcquireAppExternalURL(testAppNames[0])
+			require.NotEmpty(t, externalURL, "external URL must not be empty")
+
+			// Check if test app endpoint is available
+			require.NoError(t, utils.HealthCheckApps(externalURL))
+
+			time.Sleep(2 * time.Second)
```

Comment:
What is this sleep, and the one below, used for? Remove if unneeded, or add a comment if we end up keeping them

Reply:
Removed unnecessary sleep and added corresponding comments.